src/main/java/net/sf/jabref/logic/autosaveandbackup/AutosaveManager.java

```diff
@@ -0,0 +1,94 @@
+package net.sf.jabref.logic.autosaveandbackup;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+import net.sf.jabref.model.database.BibDatabaseContext;
+import net.sf.jabref.model.database.event.AutosaveEvent;
+import net.sf.jabref.model.database.event.BibDatabaseContextChangedEvent;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * Saves the given {@link BibDatabaseContext} on every {@link BibDatabaseContextChangedEvent} by posting a new {@link AutosaveEvent}.
+ * An intelligent {@link ExecutorService} with a {@link BlockingQueue} prevents a high load while saving and rejects all redundant save tasks.
+ */
+public class AutosaveManager {
+
+    private static final Log LOGGER = LogFactory.getLog(AutosaveManager.class);
+
+    private static Set<AutosaveManager> runningInstances = new HashSet<>();
+
+    private final BibDatabaseContext bibDatabaseContext;
+    private final BlockingQueue<Runnable> workerQueue;
+    private final ExecutorService executor;
+    private final EventBus eventBus;
+
+
+    private AutosaveManager(BibDatabaseContext bibDatabaseContext) {
+        this.bibDatabaseContext = bibDatabaseContext;
+        this.workerQueue = new ArrayBlockingQueue<>(1);
+        this.executor = new ThreadPoolExecutor(1, 1, 0, TimeUnit.SECONDS, workerQueue);
+        this.eventBus = new EventBus();
+    }
+
+    @Subscribe
+    public synchronized void listen(@SuppressWarnings("unused") BibDatabaseContextChangedEvent event) {
```

Comment:
> What is the purpose of this class?

Event classes `EntryEvent`, `MetaDataChangedEvent`, etc. extend  `BibDatabaseContextChangedEvent`. E.g., every time an `EntryEvent` is posted, the `BibDatabaseContextChangedEvent` will get posted too. So this class summarizes multiple eventclasses. Otherwise we would have to listen for each Event in separate methods.

> It looks like it just reposts the event, right? 

If you look closer, a new `AutosaveEvent` is posted within a submission to the executor. This is notifying a class within the `gui` package which actually starts `new SaveDatabaseAction(panel).runCommand();` (I'm not really happy with the logical code of `SaveDatabaseAction` as this is placed whithin a `gui` package, but we have to use it.)

Now the `executor` submits the tasks in asynchronous way while the eventBus is synchronous. Therefore a new Event is used. Of course I coud move `...runCommand();` here, but this would violate the _MVC pattern_. So semantically this does not repost the event.